tests/BlockStorage.test.ts

```diff
@@ -0,0 +1,358 @@
+        "0x6573c3cbd0a5afe60e42432790448adf2b9a242b3393a3019b5004562e64d705675bb632ebf0108716299ea9ccbe2029e653eff0456330b8d3bc042d3c8aa5da",
+        "0x89ab2ca7add018bee036168c13ced1afa1fc66ba164f4f8bbc01ab89aaf701c9e6fa427c2bb519307907d31ea2e5860bd31a265fdca0f68b633c1a3340584813",
+        "0xaf60bc9cd1ef5efae6dca87e35626748093019268dc6d7059d0891ce685ee6b1007c83e2b2d8c6326d039f5118789693667b25ac1452729e2c28442e77af5fd8",
+        "0xd9ee67e43efeedad6d57aa30c5a357f7393972dc34b0ab1e975742c1673341455165ac6ddfe7343304ec863208c6bfa6cbbbc2bb62f93dfbabc422452ddad96f",
+        "0xf1c2bf8ede3a4ae4fd25288a53a45907ec018054cc156c2251e1860924e085f5fc0b46b1a9e90a2954a6c93674e2b8673d3ec008bb434ba9119583fbb56a885c",
+        "0xd53965f0b1138bae1784106257fec91fb71c521afb8d304d2315350caf25f534e574a006c5dcf4fa2b0726edc43ce6c8f559ba51a9e6d39cdb67d303d4899efd",
+        "0x50719e87d4f08c339e26c0c55ab92677df047cdb0950d8ab3135b8fd618b3c2ba4690ef019c5442fd82d84200c71441a3d14d6fbb290d18d33926c17075ee62e",
+        "0xcf097be8bdbb562e7c5ea00e5cc6ca2151e3c807ea2d566abe191ff8e81acfd491187a975aa7ee5b568f840a46a669aeb061999d294c47944256424f1a4c5d10",
+        "0x1662d6a3901f03096c664016b6869586ee4bdcfb9268b70c4ba0c2c4837a9e2af81415744234bb1feaff2f63aea7ab18f77834b743267523257155c7fe191ff7",
+        "0x1d5a065bf2fccf78bdaa4456e96fbf0dd697ba2d6822d64a4266e862ac9e96f59e12b5234c66ec2a0e9c15a129503ffd61769d5f04b10b9c702e5d77ef6206a7",
+        "0x54522e3402cbb17e39dae09e4aa0cc42c495a6332d1ff8ef3c22b83104720a06d9ffba4c68c959f48b4bb4a3bd8a68f34f77f760ed284d31b45d77b2ce639dd8",
+        "0x9c4a20550ac796274f64e93872466ebb551ba2cd3f2f051533d07a478d2402b59e5b0f0a2a14e818b88007ec61d4a82dc9128851f43799d6c1dc0609fca1537d"
+      ]
+    }
+  ];
+
+var block_storage : BlockStorage.default;
+
+/**
+ * Creates BlockStorage
+ */
+function create ()
+{
+    block_storage = new BlockStorage.default(':memory:', putAllData);
+}
+
+/**
+ * Puts all data
+ */
+function putAllData ()
+{
+    var idx = 0;
+    var doPut = () =>
+    {
+        block_storage.put(sample_data[idx], () =>
+        {
+            idx++;
+            if (idx < sample_data.length)
+            {
+                doPut();
+            }
+            else
+            {
+                getData(1);
+                return;
+            }
+        });
+    }
+    doPut();
+}
+
+/**
+ * Gets one data
+ */
+function getData (height : number)
+{
+    var res = block_storage.get(height, (err:any, rows:any) =>
+    {
+        assert.equal(rows.length, 1);
+        assert.equal(rows[0].height, 1);
+        assert.equal(rows[0].merkle_root,
+          '0x9c4a20550ac796274f64e93872466ebb551ba2cd3f2f051533d07a478d2402b' +
+          '59e5b0f0a2a14e818b88007ec61d4a82dc9128851f43799d6c1dc0609fca1537d');
+    });
+}
+
+/**
+ *  Start test
+ */
+function runBlockStorageTest ()
+{
+    create();
+}
+
+describe('BlockStorage', () =>
+{
+    it('BlockStorage', () =>
```

Comment:
Please give it an actual description. This is supposed to read as a sentence.

Reply:
Yeah, I think we should be more specific. ex) `Tests on the creation, writing, and reading of BlockStorage.`